src/static/js/mailmerge.js

```diff
@@ -114,39 +114,50 @@ $(function() {
-                '<p class="centered"><span class="ui-icon ui-icon-info"></span>',
-                _("Valid tokens for the subject and text") + ':',
-                '<br/><br/>',
-                mailmerge.render_fields(),
-                '</p>',
-                '</div>',
-                '</td>',
-                '</tr>',
-                '</table>',
+                tableform.fields_render(
+                    [
+                        { id: 'em-from', post_field: 'from', label: _("From"), type: 'text', doublesize: true, validation: 'notblank' },
+                        { id: 'em-subject', post_field: 'subject', label: _("Subject"), type: 'text', doublesize: true, validation: 'notblank',
+                            callout: _("Valid tokens for the subject and text") + ':' +
+                            '<br/><br/>' +
+                            mailmerge.render_fields()
+                        },
+                        { id: 'em-body', post_field: 'body', label: _("Email body"), type: 'richtextarea', height: '200px', validation: 'notblank' },
+                        { id: 'em-template', label: _("Template"), type: 'select', options: edit_header.template_list_options(controller.templates) },
+                        { id: 'em-includeunsubscribe', post_field: 'unsubscribe', post_field: 'em-includeunsubscribe', label: _("Add an unsubscribe link to the bottom of emails"), type: 'check' },
+                        { id: 'em-logemail', post_field: 'logemail', label: _("Add a log to recipient person records"), type: 'check',
+                            hideif: function() {
+                                if ( controller.fields.includes("ID") || controller.fields.includes("OWNERID") ) {
+                                    return false;
+                                } else {
+                                    return true;
+                                }
+                            }
+                        },
+                        { id: 'em-logtype', post_field: 'logtype', label: _("Log type"), type: 'select', options: html.list_to_options(controller.logtypes, 'ID', 'LOGTYPENAME'),
+                            hideif: function() {
+                                if ( controller.fields.includes("ID") || controller.fields.includes("OWNERID") ) {
+                                    return false;
+                                } else {
+                                    return true;
+                                }
+                            }
+                        },
+                        { id: 'em-logmessage', post_field: 'logmessage', label: _("Log message"), type: 'textarea', validation: 'notblank',
+                            hideif: function() {
+                                if ( controller.fields.includes("ID") || controller.fields.includes("OWNERID") ) {
+                                    return false;
+                                } else {
+                                    return true;
```

Comment:
For readability, I'd put this code inside an extra function - mailmerge.log_unavailable() just in case it changes in future or if other areas need to test it.

You can then just have ```hideif: mailmerge.log_unavailable``` for these 3 UI elements.